src/rgw/rgw_data_sync.cc

```diff
@@ -1609,6 +1609,13 @@ class RGWDataSyncShardCR : public RGWCoroutine {
         drain_all();
         return set_cr_error(retcode);
       }
+      // clean up full sync index
+      yield {
+        const auto& pool = sync_env->svc->zone->get_zone_params().log_pool;
+        auto index_oid_prefix = datalog_sync_full_sync_index_prefix + "." + sc->source_zone.id;
```

Comment:
I think the function, `full_data_sync_index_shard_oid()` that builds the oid string can be reused here?

Reply:
much better! updated